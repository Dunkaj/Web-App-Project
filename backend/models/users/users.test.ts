import UserActions from './users'


var userActions=new  UserActions()

test('create a new user',async()=>{
  userActions.createUser({
    // All details that required by the interface
  })
  var [user,err]=await userActions.insert()

  expect(err).toBeFalsy()
})
z

test('create a user with missing requried parameter',()=>{
  var [user,error]=userActions.createUser({
    // All details that required by the interface
  })
  expect(error).toBeTruthy()
})

test('user.getUserById existing',async ()=>{
    var insertResult=userActions.createUser({
    })
    var getResult=userActions.getUserById(user.id)
    expect(insertResult.user.id).toEqual(getResult.user)
})

test('getUserById not existing',async ()=>{
    var getResult=userActions.getUserById(-10)
    expect(getResult.user).toBeFalsy()
    expect(getResult.error).toBeFalsy()
})

test('get a user that has been created',async ()=>{
    var insertResult=userActions.createUser({
    })
    var getResult=userActions.getUserById(user.id)
    expect(insertResult.user.id).toEqual(getResult.user)
})


test('reset User password for existing user',async ()=>{
    // Create user
    var getResult=userActions.resetPassword(user_id,newPassword)
    expect(getResult.ok).toBeTruthy()
})
test('reset User password for non- existing user',async ()=>{
    var getResult=userActions.resetPassword(-200,newPassword)
    expect(getResult.ok).toBeFalsy)
    expect(getResult.ok).toBeTruthy
})
