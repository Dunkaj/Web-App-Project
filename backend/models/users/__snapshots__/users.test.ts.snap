// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOne (remove created, id from snapshot) 1`] = `
Object {
  "access": null,
  "company": null,
  "country": null,
  "disabled": null,
  "email": "example_email@example_email.com",
  "fname": "first name",
  "lname": "last name",
  "login": null,
  "phone": null,
  "provider": null,
  "provider_id": null,
  "role": "user",
}
`;

exports[`getUserByEmailAndPassword (remove id,created,password from snapshot) 1`] = `
Object {
  "access": null,
  "company": null,
  "country": null,
  "disabled": null,
  "email": "myemail5@myemail.com",
  "fname": "",
  "lname": "",
  "login": null,
  "phone": null,
  "provider": null,
  "provider_id": null,
  "role": "user",
}
`;

exports[`setNewPassword, and getUserByEmailAndPassword (remove password,created,id from snapshot) 1`] = `
Object {
  "access": null,
  "company": null,
  "country": null,
  "disabled": null,
  "email": "myemail22@myemail.com",
  "fname": "",
  "lname": "",
  "login": null,
  "phone": null,
  "provider": null,
  "provider_id": null,
  "role": "user",
}
`;
